src/main/java/bio/terra/workspace/app/controller/WorkspaceApiController.java

```diff
@@ -276,7 +276,7 @@ private AuthenticatedUserRequest getAuthenticatedInfo() {
   public ResponseEntity<Void> grantRole(
       @PathVariable("id") UUID id,
       @PathVariable("role") IamRole role,
-      @PathVariable("memberEmail") String memberEmail) {
+      @RequestBody String memberEmail) {
```

Comment:
I wonder if the body should be JSON. That would allow upward change, if needed.
I wonder if the role should be in the path or also moved to the JSON.

The asymmetry with using DELETE is annoying. Maybe we make a PUT that does add or delete so we can have a single expression of a change rather than two different API syntaxes.

Reply:
Why not POST & DELETE?

I agree re: putting role into the body and making it a JSON object.

Reply:
There's a good discussion [here](https://stackoverflow.com/a/1233569/12345554). PATCH is another option they mention.